### PR TITLE
neon: Fix missing active account

### PR DESCRIPTION
**Pull request by provokateurin**

### CODE REVIEW

File: packages/neon/neon/lib/src/blocs/accounts.dart [hunk 1]
```diff
@@ -37,19 +37,28 @@ class AccountsBloc extends Bloc implements AccountsBlocEvents, AccountsBlocState
     });
 
     final as = accounts.value;
+
     if (_globalOptions.rememberLastUsedAccount.value && _storage.containsKey(_keyLastUsedAccount)) {
       final lastUsedAccountID = _storage.getString(_keyLastUsedAccount);
-      activeAccount.add(lastUsedAccountID != null ? as.find(lastUsedAccountID) : null);
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/blocs/accounts.dart:42]: For some reason this code failed because the key was there, but the value was null.

Comment (provokateurin) [reply packages/neon/neon/lib/src/blocs/accounts.dart:42]: I assume I was wiped here when I clicked on the notification: https://github.com/provokateurin/nextcloud-neon/blob/35eb2d65799f3e6a5f53b62da8cac868d3e52f86/packages/neon/neon/lib/src/app.dart#L143
But I don't understand how the account could be null.

File: packages/neon/neon/lib/src/blocs/accounts.dart [hunk 2]
```diff
@@ -37,19 +37,28 @@ class AccountsBloc extends Bloc implements AccountsBlocEvents, AccountsBlocState
...
-      );
+      if (lastUsedAccountID != null) {
+        final aa = as.find(lastUsedAccountID);
+        if (aa != null) {
+          activeAccount.add(aa);
+        }
+      }
     }
+
+    _globalOptions.initialAccount.stream.listen((final lastAccount) {
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/blocs/accounts.dart:51]: Why do we need to listen to this?
we only rely on the executed logic on startup but are still listening to it the for entire life cycle of the app.

**Comment by Leptopoda** [comment]
Could it be that 
https://github.com/provokateurin/nextcloud-neon/blob/35eb2d65799f3e6a5f53b62da8cac868d3e52f86/packages/neon/neon/lib/src/blocs/accounts.dart#L28

needs to be called?
Currently we load the accounts and then register the listener thus not executing it.

File: packages/neon/neon/lib/src/blocs/accounts.dart (hunk 2 above)
Comment (provokateurin) [reply packages/neon/neon/lib/src/blocs/accounts.dart:51]: I forgot to revert it to the `.stream.first.then`

File: packages/neon/neon/lib/src/blocs/accounts.dart [hunk 3]
```diff
@@ -37,19 +37,28 @@ class AccountsBloc extends Bloc implements AccountsBlocEvents, AccountsBlocState
...
-          final account = lastAccount != null ? as.find(lastAccount) : null;
-          if (account != null) {
-            activeAccount.add(account);
-          }
-        }),
-      );
+      if (lastUsedAccountID != null) {
+        final aa = as.find(lastUsedAccountID);
+        if (aa != null) {
+          activeAccount.add(aa);
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/blocs/accounts.dart:46]: Can we use setActiveAccount?

File: packages/neon/neon/lib/src/blocs/accounts.dart [hunk 4]
```diff
@@ -37,19 +37,28 @@ class AccountsBloc extends Bloc implements AccountsBlocEvents, AccountsBlocState
...
+        }
+      }
     }
+
+    _globalOptions.initialAccount.stream.listen((final lastAccount) {
+      final account = lastAccount != null ? as.find(lastAccount) : null;
+      // As long as no active account has been set it should set the active account
+      if (activeAccount.valueOrNull == null) {
+        if (account != null) {
+          activeAccount.add(account);
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/blocs/accounts.dart:56]: Can we use setActiveAccount?

File: packages/neon/neon/lib/src/blocs/accounts.dart [hunk 5]
```diff
@@ -37,19 +37,28 @@ class AccountsBloc extends Bloc implements AccountsBlocEvents, AccountsBlocState
...
+        } else if (as.isNotEmpty) {
+          activeAccount.add(as.first);
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/blocs/accounts.dart:58]: Can we use setActiveAccount?

**Comment by provokateurin** [comment]
>  Currently we load the accounts and then register the listener thus not executing it.

I don't understand what you mean. In the constructor we listen to account changes and update the global options accordingly, so I don't see a problem there?

**Comment by Leptopoda** [comment]
My bad. I thought that we are adding event's to the stream and then add a listener thus the listener would not be called as the events where already added (before we had a listener).

Behaviorsubject being a broadcast stream obviously doesn't like that and the listener will still be called.

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**